analyses/infercnv-consensus-cell-type/scripts/01_run-infercnv.R

```diff
@@ -0,0 +1,241 @@
+  "annotation_file not provided" = !is.null(opts$annotation_file),
+  "output_dir was not specified" = !is.null(opts$output_dir),
+  "hmm_model not properly specified" = opts$hmm_model %in% c("i3", "i6")
+)
+
+# Setup files and paths -----------------
+set.seed(opts$seed)
+
+# read SCE file
+sce <- readRDS(opts$sce_file)
+
+# Read in reference SCE file
+ref_sce <- readRDS(opts$reference_file)
+
+# define reference name to save in output
+ref_name <- stringr::str_remove(
+  basename(opts$reference_file),
+  "^ref-"
+) |>
+  stringr::str_remove("\\.rds$")
+
+# get library id and construct scratch directory
+library_id <- metadata(sce)$library_id
+
+# create output and scratch directory if not already present for library id
+fs::dir_create(opts$output_dir)
+
+# organize scratch by library id
+scratch_dir <- file.path(opts$scratch_dir, library_id)
+fs::dir_create(scratch_dir)
+
+# define output metadata file
+cnv_metadata_file <- file.path(opts$output_dir, glue::glue("{library_id}_cnv-metadata.tsv"))
+# define path to output cnv metadata file saved in scratch dir
+scratch_metadata_file <- file.path(scratch_dir, "map_metadata_from_infercnv.txt")
+
+# define output infercnv obj
+scratch_obj_file <- file.path(scratch_dir, "run.final.infercnv_obj")
+output_obj_file <- file.path(opts$output_dir, glue::glue("{library_id}_cnv-obj.rds"))
+
+# png file to save
+scratch_png <- file.path(scratch_dir, "infercnv.png")
+output_png <- file.path(opts$output_dir, glue::glue("{library_id}_infercnv.png"))
+
+
+# Prepare input data  ---------------------------------------------
+
+# Update SCE column names to always prefix with library for downstream bookkeeping
+colnames(sce) <- glue::glue("{library_id}-{colnames(sce)}")
+
+# Remove any reference cells which also appear in this library
+duplicated_cells <- intersect(colnames(ref_sce), colnames(sce))
+sce <- sce[, !(colnames(sce) %in% duplicated_cells)]
+
+# Create input matrix by combining library SCE and reference SCE counts assays
+raw_counts_matrix <- cbind(
+  as.matrix(counts(sce)),
+  as.matrix(counts(ref_sce))
+)
+
+# Check that we removed the duplicates:
+stopifnot(
+  "Duplicate cells present in input to inferCNV" = !all(duplicated(colnames(raw_counts_matrix)))
+)
+
+# Define reference group name for inferCNV
+reference_group_name <- "reference"
+
+# Create and export annotation table for inferCNV:
+# "unknown" cells are uncharacterized, and "reference" cells are in the reference
+data.frame(cell_id = colnames(raw_counts_matrix)) |>
+  dplyr::mutate(annotations = dplyr::if_else(cell_id %in% colnames(sce), "unknown", "reference")) |>
+  readr::write_tsv(opts$annotation_file, col_names = FALSE)
+
+
+# Run infercnv ------------------------
+
+# create the infercnv object
+infercnv_obj <- infercnv::CreateInfercnvObject(
+  raw_counts_matrix = raw_counts_matrix,
+  annotations_file = opts$annotation_file,
+  delim = "\t",
+  gene_order_file = opts$gene_order_file,
+  ref_group_names = reference_group_name
```

Comment:
```suggestion
  ref_group_name = reference_group_name
```
I think this should probably be singular?